src/RA_Core.cpp

```diff
@@ -1439,8 +1439,10 @@ void _WriteBufferToFile(const std::wstring& sFileName, const std::string& raw)
     ofile.write(raw.c_str(), ra::to_signed(raw.length()));
 }
 
-bool _ReadBufferFromFile(_Out_ std::string& buffer, const wchar_t* sFile)
+_Use_decl_annotations_
+bool _ReadBufferFromFile(std::string& buffer, const wchar_t* sFile)
 {
+    buffer = "";
```

Comment:
Since it's an "out" parameter it needs to be initialized to something immediately. I know in C# it forces you to initialize out parameters to "null" but C++ doesn't. It was caught while running code analysis using Microsoft's Recommended Native Rules. There were tons of warnings from it but ignored most of them. I was originally checking if my SAL annotations were correct.